Branch: 17557-added isAbstract to Boolean, Fixes #17557

sanyakapoor27 opened the pull request:
Fixes #17557 
-  Class Boolean returns True now on Boolean isAbstract.
- Opened a new PR to fix branch related issues and ensure cleaner commits.

sanyakapoor27 committed:
added isAbstract to Boolean

ErikOnBike commented:
Hi @sanyakapoor27
Nice to see your PR!
The issue (as well as fix) might seem pretty simple at first. You've chosen an implementation which is not uncommon in Pharo (although you could also have chosen `self == Boolean` to prevent having to access the class' name).

An interesting approach could also be, to NOT perform any checks/comparisons. We know there are exactly 2 subclasses of Boolean. Boolean's have a very interesting way to help us have something other languages call 'conditionals'. The `ifTrue:` and `ifFalse:` methods are implemented in such a way that we do not need to check for the condition itself. Could you come up with a solution without performing any actual comparison?
Just answer/put a response here, don't update the code just yet! :-)

sanyakapoor27 commented:
Hi there @ErikOnBike ! 
Thank you for your feedback! To avoid using any comparisons or checks, could we create Boolean isAbstract and have it return true while overriding it in True & False classes to return false? Please let me know if this aligns with what you're thinking of, otherwise I'll think of another solution! thanks :)

guillep committed:
Fix closing window shortcuts in combination with window groups

MarcusDenker committed:
Merge branch 'Pharo13' into fix-window-closing

estebanlm committed:
bump NewTools to v0.10.2

estebanlm committed:
Merge pull request #18204 from estebanlm/Pharo13

bump NewTools to v0.10.2

estebanlm committed:
bump NewTools to v0.10.3

Gabriel-Darbord committed:
Fix: Rubric error when jumping to previous word and string begins with separators

estebanlm committed:
Merge pull request #18207 from estebanlm/Pharo13

bump NewTools to v0.10.3

estebanlm committed:
bump iceberg to v2.4.1

estebanlm committed:
Merge pull request #18210 from estebanlm/Pharo13

bump iceberg to v2.4.1

tinchodias committed:
OSWindow-SDL2: setCursor if window is focused and when focus is gained

Fixes #18234

tinchodias committed:
Revert wrong deletion of 2 methods

MarcusDenker committed:
fixes #18225, adds empty method as described in the issue

jecisc committed:
Merge pull request #18238 from MarcusDenker/18225-13-Be-robust-against-unintentionally-sending-initialize-to-a-Trait

Fix for #18225: Be robust against unintentionally sending #initialize to a Trait (Pharo 13)

jecisc committed:
Merge pull request #18232 from Gabriel-Darbord/P13-fix-rubric-previous-word-separators-at-start

[P13 Backport] Fix: Rubric error when jumping to previous word and string begins with separators

tesonep committed:
Adding validation of the hierarchy for all the slots
Fix #18227

tesonep committed:
Supporting the case when the hierarchy is deeper.

tesonep committed:
Adding exceptions to QA rules

tesonep committed:
Using the instance variable so the check of unused instance variables does not complain

tesonep committed:
Merge branch 'Pharo13' into fix-window-closing

tesonep committed:
Merge pull request #18243 from tesonep/18227-How-to-break-a-hierarchy-P13

[Pharo13] Adding validation of the hierarchy for all the slots

tesonep committed:
Merge branch 'Pharo13' into fix-window-closing

tesonep committed:
Merge pull request #18116 from guillep/fix-window-closing

Fix closing window shortcuts in combination with window groups

tesonep committed:
Updating Roassal to a latest release fixing some issues of the menu entries

tesonep committed:
Fixing the Toplo Version

tesonep committed:
Merge pull request #18263 from pharo-project/fixing-toplo-version

Fixing the Toplo Version

tesonep committed:
Merge branch 'Pharo13' into change-roassal-version

MarcusDenker committed:
Merge pull request #18262 from tesonep/change-roassal-version

[Pharo13] Update Roassal to new Released version

MarcusDenker committed:
Merge pull request #18280 from tinchodias/FixCursorBackport

Fix cursor on multiple OSWindows

jecisc committed:
Fix Duplicate class refactoring if there are spaces in package or package tag

Currently StRequestClassPresenter is checking if the name of a class, package and package tag are right. But is expects that we have no space in the package name or package tag name. But it is possible to have spaces there in Pharo!

Update the check.

Fixes #18287

jecisc committed:
Fix protocol

jecisc committed:
Add tests on class package and tag name validation

jecisc committed:
Merge pull request #18290 from jecisc/refactorings/backport

Backport #18289 to P13: Fix Duplicate class refactoring

demarey committed:
rename examples and ClapHelp command

Ducasse committed:
Merge pull request #18312 from demarey/clap-help-fix

rename examples and ClapHelp command

estebanlm committed:
updated spec to v2.0.3 and newtools to v0.10.4

estebanlm committed:
Merge pull request #18313 from estebanlm/Pharo13

uodate new tools to v0.10.4 and spec to v2.0.3

balsa-sarenac committed:
Fix: rename method violators are list and not single instance

balsa-sarenac committed:
Fix: use correct method to find method

Ducasse committed:
Merge pull request #18363 from balsa-sarenac/fix/rename-method-violators

Fix: rename method violators are list and not single instance

tesonep committed:
Patching the process so the files are P30 and not P31

tesonep committed:
Hardcodding version for the tests

MarcusDenker committed:
Merge pull request #18379 from tesonep/continue-to-generate-P30-artifacts

continue-to-generate-P30-artifacts

MarcusDenker committed:
Merge branch 'Pharo13' into temp

jecisc commented:
We are getting way more stuff that wanted doing that :/ It would be best to redo a PR I think
